doc/pfc_asym/PFC_Asymmetric_HLD.md

```diff
@@ -0,0 +1,391 @@
+```setup, pfc_storm_template, pfc_storm_runner, deploy_pfc_gen```
+
+Description each of the fixture is below.
+
+### deploy_pfc_gen (scope="module", autouse=True)
+
+Deploy ```roles/test/files/helpers/pfc_gen.py``` to the fanout host.
+This step can be different for different platforms. Below there is description how it works for Mellanox and Arista cases, also how to add support of another platform type.
+
+**Mellanox platform**
+
+PFC packet generator is automatically deployed during fanout deployment procedure.
+
+Example of deploying fanout for Mellanox:
+```
+ansible-playbook -i lab fanout.yml -l ${FANOUT} --become --tags pfcwd_config -vvvv
+```
+
+**Arista platform**
+
+Deploy steps:
+- Ensure destination directory exists on fanout:
+"/mnt/flash/"
+- Create pfc generator file in case it doesn't exist
+"/mnt/flash/pfc_gen.py"
+- Deploy PFC generator to the fanout switch:
+Copy "roles/test/files/helpers/pfc_gen.py" to "/mnt/flash" directory
+
+**Other platforms**
+
+Tests currently support deployment of arista fanout switches, **to support other platforms:**
+
+1. Add platform specific logic to deploy pfc packet generator automatically in ```deploy_pfc_gen``` pytest fixture.
+Or manualy deploy ```roles/test/files/helpers/pfc_gen.py``` and ensure the file is available on fanout switch.
+
+2. Create ```pfc_storm_[sku].j2``` and ```pfc_storm_stop_[sku].j2``` under ```ansible/roles/test/templates/```
+to trigger pfc storm **start/stop** action.
+
+3. Set ```pfc_storm_start``` and ```pfc_storm_stop``` variables to platform-specific template names
+in ```pfc_storm_template``` pytest fixture
+
+
+### setup (scope="module")
+*Setup steps*
+
+	- Gather minigraph facts about the device
+	- Ensure topology is T0, skip tests run otherwise
+	- Get server ports OIDs
+		docker exec -i database redis-cli --raw -n 2 HMGET COUNTERS_PORT_NAME_MAP {server_ports_names}
+	- Get server ports info
+	- Get non server port info
+	- Set unique MACs to PTF interfaces
+		Run on PTF host- tests/scripts/change_mac.sh
+	- Set ARP responder:
+		- Copy ARP responder to PTF
+			src=tests/scripts/arp_responder.py; dst=/opt
+		- Copy ARP responder supervisor configuration to the PTF container
+			src=tests/scripts/arp_responder.conf.j2 dest=/etc/supervisor/conf.d/arp_responder.conf
+		- Update supervisor configuration
+			Execute on fanout:
+			supervisorctl reread
+			supervisorctl update
+	- Copy PTF tests to PTF host
+		src=roles/test/files/ptftests dest=/root
+	- Copy SAI tests to PTF host
+		src=roles/test/files/saitests dest=/root
+	- Copy PTF portmap to PTF host
+		Copy default, if existed vendor specific, copy vendor specific
+		src=ansible/roles/test/files/mlnx/default_interface_to_front_map.ini
+		dst=/root/default_interface_to_front_map.ini
+
+*Teardown steps*
+
+	- Verify PFC value is restored to default
+	- Remove SAI tests from PTF container
+	- Remove portmap
+
+*Return dictionary in format*
```

Comment:
Please explain. What is this dictionary and why is it needed?

Reply:
Fixed